setup.py

```diff
@@ -174,84 +159,53 @@ def run(self):
-                    print(f"Using newest NumPy C API for extension {ext.name}")
-                    ext.define_macros.append(DEFINE_MACRO_NUMPY_C_API)
-                else:
-                    print(
-                        f"Using old NumPy C API (version 1.7) for extension {ext.name}"
-                    )
+class build_ext_subclass(build_ext):
+    def finalize_options(self):
+        build_ext.finalize_options(self)
+        if self.parallel is None:
+            # Do not override self.parallel if already defined by
+            # command-line flag (--parallel or -j)
 
-            if sklearn._OPENMP_SUPPORTED:
-                openmp_flag = get_openmp_flag(self.compiler)
+            parallel = os.environ.get("SKLEARN_BUILD_PARALLEL")
+            if parallel:
+                self.parallel = int(parallel)
+        if self.parallel:
+            print("setting parallel=%d " % self.parallel)
 
-                for e in self.extensions:
-                    e.extra_compile_args += openmp_flag
-                    e.extra_link_args += openmp_flag
+    def build_extensions(self):
+        from sklearn._build_utils.openmp_helpers import get_openmp_flag
 
-            build_ext.build_extensions(self)
+        for ext in self.extensions:
+            if ext.name in USE_NEWEST_NUMPY_C_API:
+                print(f"Using newest NumPy C API for extension {ext.name}")
+                ext.define_macros.append(DEFINE_MACRO_NUMPY_C_API)
+            else:
+                print(f"Using old NumPy C API (version 1.7) for extension {ext.name}")
 
-    cmdclass["build_ext"] = build_ext_subclass
+        if sklearn._OPENMP_SUPPORTED:
+            openmp_flag = get_openmp_flag(self.compiler)
 
-except ImportError:
-    # Numpy should not be a dependency just to be able to introspect
-    # that python 3.8 is required.
-    pass
-
-
-def configuration(parent_package="", top_path=None):
-    if os.path.exists("MANIFEST"):
-        os.remove("MANIFEST")
+            for e in self.extensions:
+                e.extra_compile_args += openmp_flag
+                e.extra_link_args += openmp_flag
 
-    from numpy.distutils.misc_util import Configuration
-    from sklearn._build_utils import _check_cython_version
+        build_ext.build_extensions(self)
 
-    config = Configuration(None, parent_package, top_path)
-
-    # Avoid useless msg:
-    # "Ignoring attempt to set 'name' (from ... "
-    config.set_options(
-        ignore_setup_xxx_py=True,
-        assume_default_configuration=True,
-        delegate_options_to_subpackages=True,
-        quiet=True,
-    )
-
-    # Cython is required by config.add_subpackage for templated extensions
-    # that need the tempita sub-submodule. So check that we have the correct
-    # version of Cython so as to be able to raise a more informative error
-    # message from the start if it's not the case.
-    _check_cython_version()
+    def run(self):
+        self.run_command("build_clib")
```

Comment:
Running `build_clib` here is a minor convenient. This allows `python setup.py develop` to run fully from a fresh clone.